packages/shared/src/mixpanelClient.test.ts

```diff
@@ -0,0 +1,513 @@
+        mixpanelClient.trackEvent("fake-package.some-args", { foo: 123, bar: "abc" });
+
+        expect(mockMixpanelClient.track).toHaveBeenCalledTimes(2);
+        expect(mockMixpanelClient.track).toHaveBeenNthCalledWith(
+          1,
+          "fake-package.no-args",
+          {
+            distinct_id: "fake-session-id",
+            packageName: "fake-package",
+            packageVersion: "0.0.0",
+          },
+          anyCallback
+        );
+        expect(mockMixpanelClient.track).toHaveBeenNthCalledWith(
+          2,
+          "fake-package.some-args",
+          {
+            bar: "abc",
+            distinct_id: "fake-session-id",
+            foo: 123,
+            packageName: "fake-package",
+            packageVersion: "0.0.0",
+          },
+          anyCallback
+        );
+      });
+
+      describe("appendAdditionalProperties", () => {
+        it("should support appending additional default properties", async () => {
+          await initializeSession({
+            accessToken: "fake-access-token",
+            packageName: "fake-package",
+            packageVersion: "0.0.0",
+          });
+
+          mixpanelClient.trackEvent("fake-package.no-properties");
+          mixpanelClient.appendAdditionalProperties({ foo: 123 });
+          mixpanelClient.trackEvent("fake-package.some-properties");
+          mixpanelClient.appendAdditionalProperties({ bar: "abc" });
+          mixpanelClient.trackEvent("fake-package.some-more-properties");
+
+          expect(mockMixpanelClient.track).toHaveBeenCalledTimes(3);
+          expect(mockMixpanelClient.track).toHaveBeenNthCalledWith(
+            1,
+            "fake-package.no-properties",
+            {
+              distinct_id: "fake-session-id",
+              packageName: "fake-package",
+              packageVersion: "0.0.0",
+            },
+            anyCallback
+          );
+          expect(mockMixpanelClient.track).toHaveBeenNthCalledWith(
+            2,
+            "fake-package.some-properties",
+            {
+              distinct_id: "fake-session-id",
+              foo: 123,
+              packageName: "fake-package",
+              packageVersion: "0.0.0",
+            },
+            anyCallback
+          );
+          expect(mockMixpanelClient.track).toHaveBeenNthCalledWith(
+            3,
+            "fake-package.some-more-properties",
+            {
+              bar: "abc",
+              distinct_id: "fake-session-id",
+              foo: 123,
+              packageName: "fake-package",
+              packageVersion: "0.0.0",
+            },
+            anyCallback
+          );
+        });
+      });
+    });
+  });
+
+  describe("trackAsyncEvent", () => {
+    let createDeferred: typeof import("./async/createDeferred").createDeferred;
+
+    beforeEach(async () => {
+      createDeferred = require("./async/createDeferred").createDeferred;
+
+      await await initializeSession({
```

Comment:
```suggestion
      await initializeSession({
```

Reply:
Oh gosh. I got a bit out of control there didn't I.

Reply:
Better to be safe than sorry! :P